File: src/common/py/settings/default_settings.py
import typing

from ..utils.config import SettingID


def get_default_settings() -> typing.Dict[SettingID, typing.Any]:
    # pylint: disable=too-few-public-methods
    """
    Gets default values for all settings.

    Returns:
        A dictionary mapping the setting identifiers to their default values.
    """
    from .component_setting_ids import ComponentSettingIDs
    from .general_setting_ids import GeneralSettingIDs
    from .integration_setting_ids import IntegrationSettingIDs
    from .network_setting_ids import (
        NetworkSettingIDs,
        NetworkServerSettingIDs,
        NetworkClientSettingIDs,
    )

    return {
        # General settings
        GeneralSettingIDs.DEBUG: False,
        GeneralSettingIDs.DEBUG_TRACE: False,
        # Component settings
        ComponentSettingIDs.INSTANCE: "default",
        # Network settings
        NetworkSettingIDs.API_KEY: "",
        NetworkSettingIDs.VERIFY_SSL: True,
        NetworkSettingIDs.TRANSMISSION_CHUNK_SIZE: 1 * 1024 * 1024,
        NetworkSettingIDs.REGULAR_COMMAND_TIMEOUT: 10,
        NetworkSettingIDs.EXTERNAL_REQUESTS_TIMEOUT: 60,
        NetworkServerSettingIDs.ALLOWED_ORIGINS: "",
        NetworkServerSettingIDs.IDLE_TIMEOUT: 60 * 60,
        NetworkClientSettingIDs.SERVER_ADDRESS: "",
        NetworkClientSettingIDs.CONNECTION_TIMEOUT: 10,
        # Integration settings
        IntegrationSettingIDs.DEFAULT_ROOT_PATH: "/",
    }

File: src/common/web/settings/DefaultSettings.ts
import { ThemeSettings } from "../ui/theme/ThemeSettings";
import { SettingID } from "../utils/config/SettingID";
import { GeneralSettingIDs } from "./GeneralSettingIDs";
import { NetworkClientSettingIDs, NetworkSettingIDs } from "./NetworkSettingIDs";
import { ThemeSettingIDs } from "./ThemeSettingIDs";

/**
 * Gets default values for all settings.
 *
 * @returns - An object mapping the setting identifiers to their default values.
 */
export function getDefaultSettings(): Map<SettingID, any> {
    let defaults = new Map<SettingID, any>();

    // General settings
    defaults.set(GeneralSettingIDs.Debug, false);
    defaults.set(GeneralSettingIDs.SupportEmail, "sciebo.rds@uni-muenster.de");
    defaults.set(GeneralSettingIDs.VerboseNotifications, false);
    defaults.set(GeneralSettingIDs.NotificationTimeout, 3.0);

    // Network settings
    defaults.set(NetworkSettingIDs.RegularCommandTimeout, 10.0);
    defaults.set(NetworkClientSettingIDs.ServerAddress, "");
    defaults.set(NetworkClientSettingIDs.ConnectionTimeout, 10.0);

    // Theme settings
    defaults.set(ThemeSettingIDs.PrimaryColor, ThemeSettings.primaryColor);
    defaults.set(ThemeSettingIDs.LightSurfaceColor, ThemeSettings.light.surfaceColor);
    defaults.set(ThemeSettingIDs.LightHighlightColor, ThemeSettings.light.highlightColor);
    defaults.set(ThemeSettingIDs.DarkSurfaceColor, ThemeSettings.dark.surfaceColor);
    defaults.set(ThemeSettingIDs.DarkHighlightColor, ThemeSettings.dark.highlightColor);

    return defaults;
}
